wp-dependency-installer.php

```diff
@@ -84,15 +84,14 @@ public function load_hooks() {
 		 * Register wp-dependencies.json
 		 *
 		 * @param string $plugin_path Path to plugin or theme calling the framework.
+		 * @param array  $config (optional) JSON config as array.
 		 */
-		public function run( $plugin_path ) {
+		public function run( $plugin_path, $config = [] ) {
 			if ( file_exists( $plugin_path . '/wp-dependencies.json' ) ) {
 				$config = file_get_contents( $plugin_path . '/wp-dependencies.json' );
-				if ( empty( $config ) ||
-					null === ( $config = json_decode( $config, true ) )
-				) {
-					return;
-				}
+				$config = json_decode( $config, true );
+			}
+			if ( ! empty( $config ) ) {
 				$this->source = basename( $plugin_path );
```

Comment:
I agree with what you say [here](https://github.com/afragen/wp-dependency-installer/pull/44/files#r379519152), I just point out that you forgot the following point:


```php
include_once( __DIR__ . '/vendor/autoload.php' );

// Actual workflow:
// WP_Dependency_Installer::instance()->run( __DIR__ );

// Your proposed workflow:
WP_Dependency_Installer::instance()->source = basename( __DIR__ );  // <-- 😠 bad code
WP_Dependency_Installer::instance()->register( $config ); 

// Alternative workflow?
WP_Dependency_Installer::instance()->register( $config );
WP_Dependency_Installer::instance()->run( __DIR__ );
```

I have to think for a while, but it shouldn't be too much complex edit...

Reply:
Hmm, I see what you mean and your alternative workflow is correct if there is no JSON file.

I forgot about setting `$this->source` 🤦‍♂️

I think the follow is reasonable.

```php
include_once( __DIR__ . '/vendor/autoload.php' );

// Actual workflow with JSON file.
// WP_Dependency_Installer::instance()->run( __DIR__ );

// Alternative workflow using config array.
WP_Dependency_Installer::instance()->register( $config );
WP_Dependency_Installer::instance()->run( __DIR__ );
```